spec/components/proposal_details/list_component_spec.rb

```diff
@@ -3,92 +3,60 @@
 require "rails_helper"
 
 RSpec.describe ProposalDetails::ListComponent, type: :component do
-  describe "#groups" do
-    let(:proposal_details) do
-      [
-        {
-          question: "Question 1",
-          responses: [{ value: "Answer 1" }],
-          metadata: { portal_name: "Group A", auto_answered: true }
-        },
-        {
-          question: "Question 2",
-          responses: [{ value: "Answer 2" }],
-          metadata: { portal_name: "Group A" }
-        },
-        {
-          question: "Question 3",
-          responses: [{ value: "Answer 3" }],
-          metadata: { portal_name: "Group B", auto_answered: true }
-        },
-        {
-          question: "Question 4",
-          responses: [{ value: "Answer 4" }],
-          metadata: { portal_name: "Group C" }
-        }
-      ].to_json
-    end
+  let(:proposal_details) do
+    [
+      {
+        question: "Test question 1",
+        responses: [{ value: "Test response 1" }],
+        metadata: { auto_answered: true, portal_name: "group_a" }
+      },
+      {
+        question: "Test question 2",
+        responses: [{ value: "Test response 2" }],
+        metadata: { auto_answered: true, portal_name: "group_b" }
+      },
+      {
+        question: "Test question 3",
+        responses: [{ value: "Test response 3" }],
+        metadata: { auto_answered: true, portal_name: "group_a" }
+      }
+    ].to_json
+  end
+
+  let(:planning_application) do
+    build(:planning_application, proposal_details: proposal_details)
+  end
+
+  let(:component) do
+    described_class.new(
+      proposal_details: planning_application.proposal_details
+    )
+  end
+
+  before { render_inline(component) }
 
-    let(:planning_application) do
-      create(:planning_application, proposal_details: proposal_details)
-    end
+  it "renders 'Hide auto answered' check box" do
+    expect(page).to have_field(
+      "View ONLY applicant answers, hide 'Auto-answered by RIPA'"
+    )
+  end
+
+  it "renders a link for first group" do
+    expect(page).to have_link("Group a", href: "#groupa")
+  end
 
-    let(:list_component) do
-      described_class.new(
-        proposal_details: planning_application.proposal_details
-      )
-    end
+  it "renders a link for second group" do
+    expect(page).to have_link("Group b", href: "#groupb")
+  end
+
+  it "renders first group with correct numbered questions" do
+    group_a = page.find("h4", text: "Group a").find(:xpath, "../..")
+    expect(group_a).to have_content("1. \n  \n    Test question 1")
+    expect(group_a).to have_content("2. \n  \n    Test question 3")
+  end
 
-    it "returns numbered proposal details grouped by portal name" do
-      expect(list_component.send(:groups)).to eq(
```

Comment:
I think when I initially wrote these files I'd forgotten (I think I knew once!) about how to test components nicely. So I wrote a whole bunch of tests for private methods 😬.